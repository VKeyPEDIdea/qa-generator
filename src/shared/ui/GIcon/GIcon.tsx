/* eslint-disable max-len */
import { GIconProps } from './GIconProps.model';

const COLORS = {
    brown: '#986D43',
};

const GIcon = ({
	title,
	color,
    size = 24
}: GIconProps) => {
	let pathes;

	switch (title) {
		case 'house':
			pathes = (
				<>
					<path d="M0,0H24V24H0Z" fill="none"/>
					<path d="M10,20V14h4v6h5V12h3L12,3,2,12H5v8Z" fill={COLORS[color]}/>
				</>
			);
			break;
		case 'hotel':
			pathes = (
				<>
					<path d="M17,11V3H7V7H3V21h8V17h2v4h8V11ZM7,19H5V17H7Zm0-4H5V13H7Zm0-4H5V9H7Zm4,4H9V13h2Zm0-4H9V9h2Zm0-4H9V5h2Zm4,8H13V13h2Zm0-4H13V9h2Zm0-4H13V5h2Zm4,12H17V17h2Zm0-4H17V13h2Z" fill={COLORS[color]}/>
				</>
			);
			break;
        case 'edit':
            pathes = (
                <>
                    <path d="M18.85 8.65 15.475 5.3l2.4-2.375 3.35 3.375ZM3.65 20.5v-3.375L14.4 6.375l3.375 3.375L7.025 20.5Z" fill={COLORS[color]}/>
                </>
            );
            break;
        default:
			break;
	}

	return (
		<>
			<svg xmlns="http://www.w3.org/2000/svg"
				width={size}
				height={size}
				viewBox={`0 0 ${size} ${size}`}>
				{pathes}
			</svg>
		</>
	);
};

export default GIcon;