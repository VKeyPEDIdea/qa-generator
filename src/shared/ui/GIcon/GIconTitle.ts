export type GIconTitle = 'house' | 'hotel' | 'edit';