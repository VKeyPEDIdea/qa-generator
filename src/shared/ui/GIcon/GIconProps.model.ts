import Color from 'shared/types/Color.type';
import { GIconTitle } from './GIconTitle';

export interface GIconProps {
	title: GIconTitle;
	color: Color;
    size?: number;
} 