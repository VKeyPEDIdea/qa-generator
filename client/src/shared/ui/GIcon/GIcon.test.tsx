import { render, screen } from "@testing-library/react";
import { describe, expect, it } from "vitest";
import GIcon from "./GIcon";


describe('GIcon component', () => {
    it('should render brown Edit icon', () => {
        const { container } = render(<GIcon title="edit" color="brown" />);
        expect(container).toMatchSnapshot();
    });

    it('should render brown Edit icon 48px', () => {
        const { container } = render(<GIcon title="edit" color="brown" size={48} />);
        expect(container).toMatchSnapshot();
    });

    it('should render brown Hotel icon', () => {
        const { container } = render(<GIcon title="hotel" color="brown" />);
        expect(container).toMatchSnapshot();
    });

    it('should render brown House icon', () => {
        const { container } = render(<GIcon title="house" color="brown" />);
        expect(container).toMatchSnapshot();
    });
});
