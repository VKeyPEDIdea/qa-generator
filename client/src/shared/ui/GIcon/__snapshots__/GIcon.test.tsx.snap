// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`GIcon component > should render brown Edit icon 1`] = `
<div>
  <svg
    height="24"
    viewBox="0 0 24 24"
    width="24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M18.85 8.65 15.475 5.3l2.4-2.375 3.35 3.375ZM3.65 20.5v-3.375L14.4 6.375l3.375 3.375L7.025 20.5Z"
      fill="#986D43"
    />
  </svg>
</div>
`;

exports[`GIcon component > should render brown Edit icon 48px 1`] = `
<div>
  <svg
    height="48"
    viewBox="0 0 48 48"
    width="48"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M18.85 8.65 15.475 5.3l2.4-2.375 3.35 3.375ZM3.65 20.5v-3.375L14.4 6.375l3.375 3.375L7.025 20.5Z"
      fill="#986D43"
    />
  </svg>
</div>
`;

exports[`GIcon component > should render brown Hotel icon 1`] = `
<div>
  <svg
    height="24"
    viewBox="0 0 24 24"
    width="24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M17,11V3H7V7H3V21h8V17h2v4h8V11ZM7,19H5V17H7Zm0-4H5V13H7Zm0-4H5V9H7Zm4,4H9V13h2Zm0-4H9V9h2Zm0-4H9V5h2Zm4,8H13V13h2Zm0-4H13V9h2Zm0-4H13V5h2Zm4,12H17V17h2Zm0-4H17V13h2Z"
      fill="#986D43"
    />
  </svg>
</div>
`;

exports[`GIcon component > should render brown House icon 1`] = `
<div>
  <svg
    height="24"
    viewBox="0 0 24 24"
    width="24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M0,0H24V24H0Z"
      fill="none"
    />
    <path
      d="M10,20V14h4v6h5V12h3L12,3,2,12H5v8Z"
      fill="#986D43"
    />
  </svg>
</div>
`;
